src/main/java/mate/academy/dao/impl/CinemaHallDaoImpl.java

```diff
@@ -0,0 +1,59 @@
+package mate.academy.dao.impl;
+
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.CinemaHallDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.CinemaHall;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.HibernateError;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class CinemaHallDaoImpl implements CinemaHallDao {
+    @Override
+    public CinemaHall add(CinemaHall cinemaHall) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(cinemaHall);
+            transaction.commit();
+            return cinemaHall;
+        } catch (HibernateError e) {
```

Comment:
```suggestion
        } catch (HibernateException e) {
```
let's be more generic